nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,336 @@
+        Turns = [{
+          Proto = "udp";
+          URI = "turn:${TURN_DOMAIN}:${builtins.toString TURN_PORT}";
+          Username = TURN_USER;
+          Password = TURN_PASSWORD;
+        }];
+        CredentialsTTL = "12h";
+        Secret = "secret";
+        TimeBasedCredentials = false;
+      };
+      Signal = {
+        Proto = NETBIRD_SIGNAL_PROTOCOL;
+        URI = "${NETBIRD_DOMAIN}:${builtins.toString NETBIRD_SIGNAL_PORT}";
+        Username = "";
+        Password = null;
+      };
+      Datadir = "${stateDir}/data";
+      HttpConfig = {
+        Address = "127.0.0.1:${builtins.toString cfg.ports.management}";
+        AuthIssuer = NETBIRD_AUTH_AUTHORITY;
+        AuthAudience = NETBIRD_AUTH_AUDIENCE;
+        AuthKeysLocation = NETBIRD_AUTH_JWT_CERTS;
+        AuthUserIDClaim = NETBIRD_AUTH_USER_ID_CLAIM;
+        OIDCConfigEndpoint = NETBIRD_AUTH_OIDC_CONFIGURATION_ENDPOINT;
+      };
+      IdpManagerConfig = {
+        ManagerType = NETBIRD_MGMT_IDP;
+        ClientConfig = {
+          Issuer = NETBIRD_AUTH_AUTHORITY;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          ClientID = NETBIRD_IDP_MGMT_CLIENT_ID;
+          ClientSecret = NETBIRD_IDP_MGMT_CLIENT_SECRET;
+          GrantType = "client_credentials";
+        };
+        ExtraConfig = cfg.idpManagerExtraConfig;
+      };
+      DeviceAuthorizationFlow = {
+        Provider = NETBIRD_AUTH_DEVICE_AUTH_PROVIDER;
+        ProviderConfig = {
+          Audience = NETBIRD_AUTH_DEVICE_AUTH_AUDIENCE;
+          Domain = NETBIRD_AUTH0_DOMAIN;
+          ClientID = NETBIRD_AUTH_DEVICE_AUTH_CLIENT_ID;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          DeviceAuthEndpoint = NETBIRD_AUTH_DEVICE_AUTH_ENDPOINT;
+          Scope = builtins.concatStringsSep " " NETBIRD_AUTH_DEVICE_AUTH_SCOPE;
+          UseIDToken = NETBIRD_AUTH_DEVICE_AUTH_USE_ID_TOKEN;
+        };
+      };
+      PKCEAuthorizationFlow = {
+        ProviderConfig = {
+          Audience = NETBIRD_AUTH_AUDIENCE;
+          ClientID = NETBIRD_AUTH_CLIENT_ID;
+          ClientSecret = NETBIRD_AUTH_CLIENT_SECRET;
+          AuthorizationEndpoint = NETBIRD_AUTH_PKCE_AUTHORIZATION_ENDPOINT;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          Scope = builtins.concatStringsSep " " NETBIRD_AUTH_SUPPORTED_SCOPES;
+          RedirectURLs = NETBIRD_AUTH_PKCE_REDIRECT_URLS;
+          UseIDToken = NETBIRD_AUTH_PKCE_USE_ID_TOKEN;
+        };
+      };
+    };
+
+    networking.firewall.allowedTCPPorts = [ 80 443 33073 10000 ];
+
+    services.nginx.virtualHosts = {
+      ${cfg.settings.NETBIRD_DOMAIN} = {
+        forceSSL = true;
+        enableACME = true;
+
+        locations."/" = {
+          root = "${stateDir}/web-ui/";
+          tryFiles = "$uri /index.html";
+        };
+
+        locations."/signalexchange.SignalExchange/".extraConfig = ''
+          grpc_pass grpc://localhost:${builtins.toString cfg.ports.signal};
+          # grpc_ssl_verify off;
```

Comment:
remove, not remove?

Reply:
It is directly taken from https://github.com/netbirdio/netbird/blob/main/infrastructure_files/nginx.tmpl.conf, I guess we can remove it